python/cuml/common/input_utils.py

```diff
@@ -137,6 +137,9 @@ def input_to_cuml_array(X, order='F', deepcopy=False,
         else:
             X_m = CumlArray(data=X.as_gpu_matrix(order=order))
 
+    elif isinstance(X, CumlArray):
+        X_m = X
```

Comment:
Not needed right now, but could you open a github issue to make this more robust? The way to do that might be to recursively call `input_to_cuml_array` with `X.to_output(‘cupy’)`. That way if someone calls `input_to_cuml(cumlary, order=different_order_than_cumlary)` for example, they would get a new cumlarray with the intended order. I think nothing in the code requires that currently, but might be good to create the issue already just in case. 

Besides that, could you add a pytest for `input_to_cuml_array(cumlary)`?

Reply:
I think this already work as intended, but please correct me if I missed something. Here `X_m = X` makes `X_m` a CumlArray but doesn't  return right away. The usual checks of input_to_cuml_array are performed, one of them is: https://github.com/rapidsai/cuml/blob/21bb830eed6449f84ca1722e3acbb652980f522f/python/cuml/common/input_utils.py#L190-L199
I believe this line `X_m = rmm_cupy_ary(cp.array, X_m, copy=False, order=order)` will fix the order if needed.

I'm adding a pytest for this, good idea!

Reply:
you are correct